plugins/techdocs-backend/src/service/router.ts

```diff
@@ -58,14 +59,20 @@ export async function createRouter({
     const { '0': path } = req.params;
     const entityName = getEntityNameFromUrlPath(path);
 
-    publisher
-      .fetchTechDocsMetadata(entityName)
-      .then(techdocsMetadataJson => {
-        res.send(techdocsMetadataJson);
-      })
-      .catch(reason => {
-        res.status(500).send(`Unable to get Metadata. Reason: ${reason}`);
-      });
+    try {
+      const techdocsMetadata: TechDocsMetadata = await publisher.fetchTechDocsMetadata(
+        entityName,
+      );
+
+      res.send(techdocsMetadata);
+    } catch (err) {
+      logger.info(
+        `Unable to get techdocs metadata for ${entityName} with error ${err}`,
+      );
+      throw new Error(
+        `Unable to get techdocs metadata for ${entityName} with error ${err}`,
+      );
+    }
```

Comment:
I don't know what happens when we throw an error, but perhaps we should keep it the old way to send a server error as response?
```
        res.status(500).send(message);

```

Reply:
Done! I slightly modified the message, let me know if you'd like it changed back, no worries on my side :) thought I'd give a bit more context to make debugging easier.